src/System.IO.FileSystem/src/System/IO/FileSystem.Unix.cs

```diff
@@ -12,6 +12,32 @@ internal static partial class FileSystem
     {
         internal const int DefaultBufferSize = 4096;
 
+        private static bool CopyDanglingSymlink(string sourceFullPath, string destFullPath)
+        {
+            // Check if the source is a dangling symlink. In those cases, we just want to copy the link
+            Interop.Sys.FileStatus ignored;
+            if (! (Interop.Sys.Stat(sourceFullPath, out ignored) < 0 &&
+                Interop.Sys.LStat(sourceFullPath, out ignored) == 0))
```

Comment:
Nit: we can use discards instead of ignored, e.g.
```C#
if (Interop.Sys.Stat(sourceFullPath, out _) == 0 ||
    Interop.Sys.LStat(sourceFullPath, out _) != 0)
{
    return false;
}
```